### DISCUSSION TITLE
Tropical Zodiac - calculating the ascendant

### DISCUSSION THREAD

**Discussion by jorjun** [cosinekitty/astronomy#166] [Q&A] [answered]
There's a great video below explaining the concept of Greenwich Hour Angle to Aries GHA♈ 

https://www.youtube.com/watch?v=bAk-Gq5QobA

I think the ascendant sign is really a version of this greenwich hour angle at time of birth but adjusted for longitude of birth. But I don't know how to get this GHAA yet.

I will publish a link to a new zodiac tool here when I have finished an early version. If anyone is interested.

**Comment by cosinekitty** [comment, marked as answer]
Hi @jorjun. The `sidereal_time` function will return the hour angle at the Greenwich prime meridian. If you call that function and then add the east longitude of the observer, is that the number you are looking for? Just clarifying the goal before I work on this!

**Reply by jorjun** [reply]
Yes please, I still don't know exactly how to calculate the ascendant, which is supposed to be the zodiac degree on the eastern horizon, but I am certain that sidereal time is part of it, since I know that this sign changes every two hours.

**Reply by cosinekitty** [reply]
@jorjun Would you try the new [SiderealTime](https://github.com/cosinekitty/astronomy/tree/master/source/js#siderealtimedate--number) function and let me know how it works for you?

**Reply by jorjun** [reply]
Looks like the type for  SiderealTime is missing.. (using typescript)

**Reply by cosinekitty** [reply]
I see this in `astronomy.d.ts`:

```TypeScript
export declare function SiderealTime(date: FlexibleDateTime): number;
```

Are you getting a TypeScript error trying to use it? Not sure what you are seeing.

**Reply by jorjun** [reply]
I see what you see in github source latest but it's not in the official package [2.0.13]

**Reply by jorjun** [reply]
Sorry. I guess you wanted me to check from github.. trying that now

**Reply by cosinekitty** [reply]
Ah, yes, now I understand... you were using npm. That's right, I haven't updated that package yet.

**Reply by jorjun** [reply]
For March 20th, 2022 15:45 GMT (Unix TS given below) I am getting an GMT hour angle of 54, should this be close to 0 for the vernal equinox?

{gah: 54.40760931521784, ts: 1647791100}

**Reply by cosinekitty** [reply]
No, because the sidereal hour angle isn't related to the Earth's orbital position. It is based only on the rotation of the Earth. You can think of it as a clock that goes from 0 to 24 hours, but they are sidereal hours, not solar hours. If you are standing in Greenwich, the sidereal time is 0 every time the equinox point crosses the meridian (the imaginary line in the sky that passes directly overhead from north to south).

**Reply by jorjun** [reply]
Everything looks good then. I am still on a quest to find the correct horoscope ascendant, but I think I am close. And this new feature really helps, thanks.
